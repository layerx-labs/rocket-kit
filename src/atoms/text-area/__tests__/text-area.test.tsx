import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { TextArea } from '../../..';

describe('TextArea', () => {
  it('renders', () => {
    const { asFragment } = render(<TextArea value="this is my text" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('has correct value', () => {
    const textValue = 'this is my text';
    render(<TextArea value={textValue} />);
    const foundInput = screen.getByDisplayValue(textValue);
    expect(foundInput).toHaveValue(textValue);
  });

  it('has empty value', () => {
    render(<TextArea />);
    const foundInput = screen.getByRole('textbox');
    expect(foundInput).toHaveValue('');
  });

  it('has correct placeholder', () => {
    const placeholderValue = 'this is my placeholder';
    render(<TextArea placeholder={placeholderValue} />);
    const foundInput = screen.getByPlaceholderText(placeholderValue);
    expect(foundInput).toHaveProperty('placeholder', placeholderValue);
  });

  it('has empty placeholder', () => {
    render(<TextArea />);
    const foundInput = screen.getByRole('textbox');
    expect(foundInput).toHaveProperty('placeholder', '');
  });
});
