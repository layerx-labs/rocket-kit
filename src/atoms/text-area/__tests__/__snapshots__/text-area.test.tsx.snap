// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea renders 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
}

.c1 {
  border-width: 0 0 0.0625rem 0;
  border-style: solid;
  border-color: #939393;
  border-radius: 0;
  width: 100%;
  height: 6.25rem;
  min-height: rem('40px');
  padding: 0;
  font-family: inherit;
  font-size: 1rem;
  word-break: break-word;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  resize: none;
}

.c1:focus {
  border-color: #4329A6;
  outline: none;
}

.c1:disabled {
  background-color: #c4c7d3;
  color: #939393;
}

.c1:invalid {
  border-color: #ef5766;
  outline: none;
  box-shadow: none;
  -webkit-box-shadow: none;
  -moz-box-shadow: none;
}

.c1:required {
  border-color: #939393;
}

.c1::-webkit-input-placeholder {
  color: #939393;
  opacity: 1;
}

.c1::-moz-placeholder {
  color: #939393;
  opacity: 1;
}

.c1:-ms-input-placeholder {
  color: #939393;
  opacity: 1;
}

.c1::placeholder {
  color: #939393;
  opacity: 1;
}

.c1:-ms-input-placeholder {
  color: #939393;
  opacity: 1;
}

.c1::-ms-input-placeholder {
  color: #939393;
  opacity: 1;
}

<div
    class="c0 text-field"
  >
    <textarea
      class="c1 text-area"
    >
      this is my text
    </textarea>
  </div>
</DocumentFragment>
`;
